otx/core/data/adapter/classification_dataset_adapter.py

```diff
@@ -120,3 +127,31 @@ def _generate_classification_label_schema(
 
     def _select_data_type(self, data_candidates: Union[list, str]) -> str:
         return "imagenet" if "imagenet" in data_candidates else data_candidates[0]
+
+
+class SelfSLClassificationDatasetAdapter(ClassificationDatasetAdapter):
```

Comment:
Could you add unit-test for this new class?

Reply:
Ok

Reply:
Done